test/e2e/clusterclass_changes.go

```diff
@@ -459,3 +466,43 @@ func rebaseClusterClassAndWait(ctx context.Context, input rebaseClusterClassAndW
 	Expect(afterControlPlane.GetGeneration()).To(Equal(beforeControlPlane.GetGeneration()),
 		"ControlPlane generation should not be incremented during the rebase because ControlPlane should not be affected.")
 }
+
+// deleteMachineDeploymentTopologyAndWaitInput is the input type for deleteMachineDeploymentTopologyAndWaitInput.
+type deleteMachineDeploymentTopologyAndWaitInput struct {
+	ClusterProxy              framework.ClusterProxy
+	Cluster                   *clusterv1.Cluster
+	WaitForMachineDeployments []interface{}
+}
+
+// deleteMachineDeploymentTopologyAndWait deletes a MachineDeploymentTopology from the Cluster and waits until the changes
+// are rolled out by ensuring the associated MachineDeployment is correctly deleted.
+func deleteMachineDeploymentTopologyAndWait(ctx context.Context, input deleteMachineDeploymentTopologyAndWaitInput) {
+	Expect(ctx).NotTo(BeNil(), "ctx is required for deleteMachineDeploymentTopologyAndWait")
+	Expect(input.ClusterProxy).ToNot(BeNil(), "Invalid argument. input.ClusterProxy can't be nil when calling deleteMachineDeploymentTopologyAndWait")
+	Expect(input.Cluster).ToNot(BeNil(), "Invalid argument. input.Cluster can't be nil when calling deleteMachineDeploymentTopologyAndWait")
+	Expect(len(input.Cluster.Spec.Topology.Workers.MachineDeployments)).To(BeNumerically(">", 0),
+		"Invalid Cluster. deleteMachineDeploymentTopologyAndWait requires at least one MachineDeploymentTopology to be defined in the Cluster topology")
+
+	log.Logf("Removing MachineDeploymentTopology from the Cluster Topology.")
+	patchHelper, err := patch.NewHelper(input.Cluster, input.ClusterProxy.GetClient())
+	// Remove the first MachineDeploymentTopology under input.Cluster.Spec.Topology.Workers.MachineDeployments
+	mdTopologyToDelete := input.Cluster.Spec.Topology.Workers.MachineDeployments[0]
+	input.Cluster.Spec.Topology.Workers.MachineDeployments = append(input.Cluster.Spec.Topology.Workers.MachineDeployments[:0], input.Cluster.Spec.Topology.Workers.MachineDeployments[0+1:]...)
```

Comment:
```suggestion
	input.Cluster.Spec.Topology.Workers.MachineDeployments = input.Cluster.Spec.Topology.Workers.MachineDeployments{1:]
```
would this work as well?